portability-spi-cloud/src/main/java/org/dataportabilityproject/spi/cloud/storage/JobStore.java

```diff
@@ -0,0 +1,55 @@
+package org.dataportabilityproject.spi.cloud.storage;
+
+import org.dataportabilityproject.spi.cloud.types.PortabilityJob;
+import org.dataportabilityproject.types.transfer.models.DataModel;
+
+import java.io.InputStream;
+
+/**
+ * Implementations handle storage and retrieval of transfer job data.
+ *
+ * This class is intended to be implemented by extensions that support storage in various back-end services.
+ */
+public interface JobStore {
+
+    /**
+     * Creates a new job.
+     */
+    void create(PortabilityJob job);
+
+    /**
+     * Updates and existing job.
+     *
+     * REVIEW: Maybe there should just be state transition-specific methods, e.g cancel(), complete(), etc.?
```

Comment:
I think those could be added but we would still need an update, particularly to capture the intermediate states in the auth flow https://github.com/google/data-portability/blob/0769409e60636d7327b68de1629449050eaaf94c/portability-core/src/main/java/org/dataportabilityproject/job/JobDao.java#L46